### ISSUE TITLE
Move `jupyter-lsp` package into Jupyter Server org

### ISSUE THREAD

**Issue by trungleduc** [jupyter-lsp/jupyterlab-lsp#805]
## What are you trying to do?

Hi @krassowski  and @bollwyvl, after our discussion about the server extension package of `jupyterlab-lsp`, we obtained the green light to move `jupyter-lsp` into Jupyter Server org. I'm going to extract this package with all commit history and move it to https://github.com/jupyter-server/jupyter-lsp.
Please let me know if there is anything I need to take care of before the moving process.

**Comment by krassowski** [comment]
:+1: I think that we would want to merge PRs: #636, #731, #778 before we start diverging the codebases. #636 would benefit from code review. The remaining two need some finishing up.

CI may need some attention as well, we would need to split responsibilities but also there may be regressions to fix.

I will have some time this weekend to look at  #731, #778 but not sure how far I will get with them.

Finally we will want to move issues related to the server side. It could involve moving https://github.com/jupyter-server/jupyter-lsp to https://github.com/jupyter-lsp organisation for half an hour, transferring issues and then moving it back to https://github.com/jupyter-server/. @jtpio did you succeed with this approach with retrolab/notebook or decide not to go for it?

**Comment by jtpio** [comment]
Yes moving the repo around went quite well and was probably the simplest approach: https://github.com/jupyter/notebook-team-compass/issues/9#issuecomment-1111013024

**Comment by trungleduc** [comment]
Thanks for the insight, I tested the moving process with the current `master`. I will redo it again once these PRs get in.

**Comment by trungleduc** [comment]
> Yes moving the repo around went quite well and was probably the simplest approach: [jupyter/notebook-team-compass#9 (comment)](https://github.com/jupyter/notebook-team-compass/issues/9#issuecomment-1111013024)

Great, I'm starting the transfer now

**Comment by krassowski** [comment]
@trungleduc are you still interested in extracting out jupyter-lsp out to another repository within jupyter-lsp as we discussed in https://github.com/jupyter-server/team-compass/issues/30? I am still happy to go with this, as long as I get help with transition/rewriting the integration tests so that we do no loose out on this. No worries if not, in that case let's just close this issue and remove the other copy of the repo.

Also apologies that I did not reply to your question (but to be fair it was added in an edit so when I read your comment first time I did not see it); unfortunately I do not have rights on jupyter-server so I cannot transfer that repository to jupyter-lsp org.

**Comment by trungleduc** [comment]
Hi @krassowski
>  in that case let's just close this issue and remove the other copy of the repo.

Indeed I think this is the way to move forward, it would save maintenance resources.
